core/src/com/unciv/logic/automation/unit/SpecificUnitAutomation.kt

```diff
@@ -493,27 +494,78 @@ object SpecificUnitAutomation {
+                tileToNuke = it
             }
         }
+        if (highestTileNukeValue > 0) {
+            Battle.NUKE(MapUnitCombatant(unit), tileToNuke!!)
+        }
         tryRelocateToNearbyAttackableCities(unit)
     }
 
+    /**
+     * Ranks the tile to nuke based off of all tiles in it's blast radius
+     * By default the value is -500 to prevent inefficient nuking.
+     */
+    fun getNukeLocationValue(nuke: MapUnit, tile: Tile): Int {
+        val civ = nuke.civ
+        if (!Battle.mayUseNuke(MapUnitCombatant(nuke), tile)) return Int.MIN_VALUE
+        val blastRadius = nuke.getNukeBlastRadius()
+        val tilesInBlastRadius = tile.getTilesInDistance(blastRadius)
+        val civsInBlastRadius = tilesInBlastRadius.mapNotNull { it.getOwner() } +
+            tilesInBlastRadius.mapNotNull { it.getFirstUnit()?.civ }
+        
+        // Don't nuke if it means we will be declaring war on someone!
+        if (civsInBlastRadius.any { it != civ && !it.isAtWarWith(civ) }) return -100000
+        // If there are no enemies to hit, don't nuke
+        if (!civsInBlastRadius.any { it.isAtWarWith(civ) }) return -100000
+        
+        // Launching a Nuke uses resources, therefore don't launch it by default
+        var explosionValue = -500
+        
+        // Returns either ourValue or thierValue depending on if the input Civ matches the Nuke's Civ
+        fun evaluateCivValue(targetCiv: Civilization, ourValue: Int, theirValue: Int): Int {
+            if (targetCiv == civ) // We are nuking something that we own!
+                return ourValue
+            return theirValue // We are nuking an enemy!
+        }
+        for (targetTile in tilesInBlastRadius) {
+            // We can only account for visible units
+            if (tile.isVisible(civ)) {
+                if (targetTile.militaryUnit != null && !targetTile.militaryUnit!!.isInvisible(civ))
+                    explosionValue += evaluateCivValue(targetTile.militaryUnit?.civ!!, -150, 50)
+                if (targetTile.civilianUnit != null && !targetTile.civilianUnit!!.isInvisible(civ))
+                    explosionValue += evaluateCivValue(targetTile.civilianUnit?.civ!!, -100, 25)
+            }
+            // Never nuke our own Civ, don't nuke single enemy civs as well
+            if (targetTile.isCityCenter() 
+                && !(targetTile.getCity()!!.health <= 50f 
+                    && targetTile.neighbors.any {it.militaryUnit?.civ == civ})) // Prefer not to nuke cities that we are about to take
+                explosionValue += evaluateCivValue(targetTile.getCity()?.civ!!, -100000, 250)
+            else if (targetTile.owningCity != null) {
+                val owningCiv = targetTile.owningCity?.civ!!
+                // If there is a tile to add fallout to there is a 50% chance it will get fallout
+                if (!(tile.isWater || tile.isImpassible() || targetTile.terrainFeatures.any { it == "Fallout" }))
```

Comment:
Is fallout parametrized in the nuke unique? I think it is, in which means this should not be a hardcoded "fallout" string

Reply:
I am not sure, lines 905 and 906 in Battle.kt use "Fallout". Fallout uses UniqueType.NullifyYields though which might have been used to abstract it from the effect.